java/.gitignore

```diff
@@ -0,0 +1,7 @@
+.gradle
+/build
+
+!gradle-wrapper.jar
```

Comment:
Can you explain this?

Also having big opaque blobs is a little scary—can we indicate the URL it came from, the SHA where we downloaded it, the version, etc?

Reply:
yeah that's actually pretty normal for java projects. I'll add the gradle wrapper validation action to this. But more info here: https://github.com/gradle/wrapper-validation-action